src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleThreadContext.java

```diff
@@ -189,36 +169,32 @@ void addImport(
             exportedName, extensionName, context.repoNameUsages.get(collisionRepoName).where());
       }
       imports.put(localRepoName, exportedName);
-      if (devDependency) {
-        devImports.add(exportedName);
-      }
     }
 
     void addTag(Tag tag) {
       tags.add(tag);
     }
 
     ModuleExtensionUsage buildUsage() throws EvalException {
+      var proxies = proxyBuilders.stream().map(p -> p.build()).collect(toImmutableList());
       var builder =
           ModuleExtensionUsage.builder()
               .setExtensionBzlFile(extensionBzlFile)
               .setExtensionName(extensionName)
               .setUsingModule(context.getModuleBuilder().getKey())
-              .setLocation(location)
-              .setImports(ImmutableBiMap.copyOf(imports))
-              .setDevImports(devImports.build())
-              .setHasDevUseExtension(hasDevUseExtension)
-              .setHasNonDevUseExtension(hasNonDevUseExtension)
+              .setProxies(proxies)
               .setTags(tags.build());
       if (isolate) {
-        if (exportedName == null) {
+        ModuleExtensionUsage.Proxy firstProxy = proxies.getFirst();
```

Comment:
Also rename to `onlyProxy` and use `Iterables.getOnlyElement`?

Reply:
done